src/sagemaker/spark/processing.py

```diff
@@ -733,9 +733,9 @@ class PySparkProcessor(_SparkProcessorBase):
 
     def __init__(
         self,
-        role: str,
-        instance_type: Union[str, PipelineVariable],
-        instance_count: Union[int, PipelineVariable],
+        role: str = None,
```

Comment:
Can you update docstring to show that these now default to None, and a quick statement of what happens when None? See `dependency_location` docstring on L781 for example